process/block/baseProcess.go

```diff
@@ -1114,7 +1116,7 @@ func (bp *baseProcessor) getLastNotarizedHdrs() []bootstrapStorage.BootstrapHead
 	}
 	bp.mutNotarizedHdrs.RUnlock()
 
-	return lastNotarizedHdrs
+	return bootstrapStorage.TrimHeaderInfoSlice(lastNotarizedHdrs)
```

Comment:
indexing instead of append could be used, than there is no need to trimheaderInfoSlice.